From: Johannes Kauffmann <johanneskauffmann@hotmail.com>
Date: Tue, 19 Nov 2024 23:30:23 +0100
Subject: musicbrainz: fix leak

(cherry picked from commit 2edddb01da361d1040736d7fda637e118ea6aee1) (edited)
edited:
- the 3.0 uses the other json parser which outputs the result differently
---
 modules/misc/webservices/musicbrainz.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/misc/webservices/musicbrainz.c b/modules/misc/webservices/musicbrainz.c
index 2393a73..2c35f75 100644
--- a/modules/misc/webservices/musicbrainz.c
+++ b/modules/misc/webservices/musicbrainz.c
@@ -56,7 +56,11 @@ static musicbrainz_lookup_t * musicbrainz_lookup(vlc_object_t *p_obj, const char
     {
         p_lookup->root = json_parse_document(p_obj, p_buffer);
         if (!p_lookup->root)
+        {
             msg_Dbg(p_obj, "No results");
+            musicbrainz_lookup_release(p_lookup);
+            p_lookup = NULL;
+        }
     }
     free(p_buffer);
     return p_lookup;
